api/pkg/file/service.go

```diff
@@ -115,7 +116,17 @@ func (s *APIService) ListFileShare(request *restful.Request, response *restful.R
 
 	ctx := common.InitCtxWithAuthInfo(request)
 
-	listFileShareRequest := s.prepareFileShareRequest(request, response)
+	backendId := request.QueryParameter(common.REQUEST_PATH_BACKEND_ID)
+
+	var listFileShareRequest *file.ListFileShareRequest
+
+	if backendId != "" {
```

Comment:
Hi @himanshuvar can we just add 'backend' at line#95, instead of checking backendId here?

Reply:
@sfzeng Adding this filter at line 95 doesn't make sense for Listing FileShare for all backends.
This filters will only applicable for getting FS for a specific backend type.

I hope this is clear.

You can refer the existing multi-cloud code.

https://github.com/sodafoundation/multi-cloud/blob/master/api/pkg/backend/service.go
Though the existing code doesn't support all other filters for the backend filer by Id .

I tried to make FS code to work with other filters too.
I hope this make sense.

Reply:
IMO, backend is similar as other filters, like 'status', 'type' and 'name', use prepareFileShareRequest instead of listFileShareByBackend and listFileShareDefault, and add 'backendid' to line#95 can achive the same target and make the code clean. So as this PR blocked others, I just created an issue to trace this: https://github.com/sodafoundation/multi-cloud/issues/1023.

Reply:
@sfzeng 
Can you please list the issues with the current code functionality wise.

I still feel that the current code is clean and have more readability. The functionalities are also separated by the functions.

IMHO listFileShareByBackend and listFileShareDefault can't achieve the same functionality. Try to understand. checkBackendExistence doesn't make sense for listFileShareDefault.
So the functonalities can not be messed up.